instrumentation/hibernate/hibernate-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v4_0/SessionTest.java

```diff
@@ -0,0 +1,880 @@
+                    span.hasKind(CLIENT)
+                        .hasParent(trace.getSpan(1))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "h2"),
+                            equalTo(SemanticAttributes.DB_NAME, "db1"),
+                            equalTo(SemanticAttributes.DB_USER, "sa"),
+                            equalTo(SemanticAttributes.DB_CONNECTION_STRING, "h2:mem:"),
+                            satisfies(
+                                SemanticAttributes.DB_STATEMENT,
+                                val -> val.isInstanceOf(String.class)),
+                            satisfies(
+                                SemanticAttributes.DB_OPERATION,
+                                val -> val.isInstanceOf(String.class)),
+                            equalTo(SemanticAttributes.DB_SQL_TABLE, "Value")),
+                span ->
+                    span.hasName("Transaction.commit")
+                        .hasKind(INTERNAL)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(
+                                stringKey("hibernate.session_id"),
+                                trace
+                                    .getSpan(1)
+                                    .getAttributes()
+                                    .get(stringKey("hibernate.session_id"))))));
+  }
+
+  private static Stream<Arguments> provideArgumentsStateQuery() {
+    return Stream.of(
+        Arguments.of(
+            named(
+                "createQuery",
+                ((Consumer<Session>) session -> session.createQuery("from Value").list())),
+            Arguments.of(
+                named(
+                    "getNamedQuery",
+                    ((Consumer<Session>)
+                        session -> session.getNamedQuery("TestNamedQuery").list())),
+                Arguments.of(
+                    named(
+                        "createSQLQuery",
+                        (Consumer<Session>)
+                            session -> session.createSQLQuery("SELECT * FROM Value").list())))));
+  }
+
+  @Test
+  void testHibernateOverlappingSessions() {
+    testing.runWithSpan(
+        "overlapping Sessions",
+        () -> {
+          Session session1 = sessionFactory.openSession();
+          session1.beginTransaction();
+
+          StatelessSession session2 = sessionFactory.openStatelessSession();
+          Session session3 = sessionFactory.openSession();
+
+          Value value1 = new Value("Value 1");
+          session1.save(value1);
+          session2.insert(new Value("Value 2"));
+          session3.save(new Value("Value 3"));
+          session1.delete(value1);
+
+          session2.close();
+          session1.getTransaction().commit();
+          session1.close();
+          session3.close();
+        });
+
+    testing.waitAndAssertTraces(
```

Comment:
Can you add a couple of `AtomicReference<String>` for the session ids, and implement the `sessionId1 != sessionId2 != sessionId3` assertion? That one assertions is the main reason this test case exists; since we need to verify that the ids don't get mixed up.